sdk/identity/azure-identity/src/main/java/com/azure/identity/EnvironmentCredential.java

```diff
@@ -56,19 +59,29 @@
         String certPath = configuration.get(Configuration.PROPERTY_AZURE_CLIENT_CERTIFICATE_PATH);
         String username = configuration.get(Configuration.PROPERTY_AZURE_USERNAME);
         String password = configuration.get(Configuration.PROPERTY_AZURE_PASSWORD);
+        logAvailableEnvironmentVariables(clientId, tenantId, clientSecret, certPath, username, password);
         if (verifyNotNull(clientId)) {
             if (verifyNotNull(tenantId, clientSecret)) {
+                logger.info("Azure Identity => EnvironmentCredential invoking ClientSecretCredential");
                 targetCredential = new ClientSecretCredential(tenantId, clientId, clientSecret, identityClientOptions);
             } else if (verifyNotNull(tenantId, certPath)) {
+                logger.info("Azure Identity => EnvironmentCredential invoking ClientCertificateCredential");
                 targetCredential = new ClientCertificateCredential(tenantId, clientId, certPath,
                         null, identityClientOptions);
             } else if (verifyNotNull(username, password)) {
+                logger.info("Azure Identity => EnvironmentCredential invoking UsernamePasswordCredential");
                 targetCredential = new UsernamePasswordCredential(clientId,
                         tenantId,
                         username,
                         password,
                         identityClientOptions);
+            } else {
+                logger.error("Azure Identity => ERROR in EnvironmentCredential: Not enough environment variables "
+                    + " to determine the environment.");
```

Comment:
Just a question about this, will there be information in the log about the environment variables that need to be set? (Like the error explaining the reason the environment credential failed?) Because this seems to me that is does not include that information, but I'm not sure if it is included later on.

Reply:
It does not. There are multiple possibilities though for a valid EnvironmentCredential. I cannot tell which scenario the user wants thus I cannot give suggestions on what environment variables are missing.

Reply:
I have completely redone the logging logic around `EnvironmentCredential`. I'm doing the best I can to guess which credential the user might be attempting to use (usually only one env var is missing from getting it to work). When multiple are missing, I print a log containing the possible combinations of env vars to configure a valid EnvironmentCredential.

Reply:
Ok that sounds good about logging the possible combination of env vars to configure a valid EnvironmentCredential. I also want to improve the logic in Go to do this. So that sounds good to me.